.github/workflows/sonarqube.yml

```
@@ -0,0 +1,75 @@
+  pull-requests: read # allows SonarQube to decorate PRs with analysis results
+
+jobs:
+  Analysis:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+      
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
```

actionlint (1.7.4)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)